rerun_py/rerun_sdk/rerun/_unions.py

```diff
@@ -14,25 +14,22 @@ def build_dense_union(data_type: pa.DenseUnionType, discriminant: str, child: pa
 
     If the discriminant string doesn't match any possible value, a `ValueError` is raised.
     """
+
+    idx = [f.name for f in list(data_type)].index(discriminant)
+    type_ids = pa.array([idx] * len(child), type=pa.int8())
+    value_offsets = pa.array(range(len(child)), type=pa.int32())
+
+    children = [pa.nulls(0, type=f.type) for f in list(data_type)]
     try:
-        idx = [f.name for f in list(data_type)].index(discriminant)
-        type_ids = pa.array([idx] * len(child), type=pa.int8())
-        value_offsets = pa.array(range(len(child)), type=pa.int32())
-
-        children = [pa.nulls(0, type=f.type) for f in list(data_type)]
-        try:
-            children[idx] = child.cast(data_type[idx].type, safe=False)
-        except pa.ArrowInvalid:
-            # Since we're having issues with nullability in union types,
-            # the cast sometimes fails but can be skipped.
-            children[idx] = child
-
-        return pa.Array.from_buffers(
-            type=data_type,
-            length=len(child),
-            buffers=[None, type_ids.buffers()[1], value_offsets.buffers()[1]],
-            children=children,
-        )
-
-    except ValueError as e:
-        raise ValueError(e.args)
```

Comment:
this whole diff is weird but i basically just removed that weird catch/raise thing